### PR TITLE
[3370][15.0][ADD] ir_attachment_image_conversion

### PR DESCRIPTION
**Pull request by kanda999**
[3370](https://www.quartile.co/web?debug=1#action=771&active_id=2413&model=project.task&view_type=list&menu_id=505)

### CODE REVIEW

**Comment by AungKoKoLin1997** [comment]
@kanda999 Please follow up the updates.

File: ir_attachment_conversion_queue_job/models/ir_attachment.py [hunk 1]
```diff
@@ -34,3 +34,30 @@ def convert_image_attachments(self, attachment):
                 attachment.res_id,
                 repr(e),
             )
+
+    def _product_image_job_queue(self):
+        mapping = {
+            "product.image": "image",
+            "product.public.category": "image",
+        }
```
Comment (yostashiro) [inline ir_attachment_conversion_queue_job/models/ir_attachment.py]: I think we might as well include the images of the partners/users.

How about images on the website?

File: ir_attachment_conversion_queue_job/models/ir_attachment.py [hunk 2]
```diff
@@ -49,11 +28,10 @@ def _product_image_job_queue(self):
                 ]
             )
             for attachment in attachments:
-                self.with_delay().convert_product_image_attachments(model, attachment)
+                self.with_delay()._convert_image_attachments(model, attachment)
```
Comment (yostashiro) [inline ir_attachment_conversion_queue_job/models/ir_attachment.py]: Add description param in `with_delay()`.

File: ir_attachment_conversion_queue_job/models/ir_attachment.py [hunk 3]
```diff
@@ -11,34 +11,13 @@
...
+    def _image_job_queue(self):
         mapping = {
+            "product.template": "image",
             "product.image": "image",
             "product.public.category": "image",
+            "res.partner": "image",
+            "res.users": "image",
         }
         for model, field in mapping.items():
             attachments = self.search(
```
Comment (yostashiro) [inline ir_attachment_conversion_queue_job/models/ir_attachment.py]: Shouldn't the search condition be adjusted to exclude already processed ones?

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: ir_attachment_conversion_queue_job/models/ir_attachment.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    def _convert_image_attachments(self, model, attachment):
+        try:
+            self.env[model].browse(attachment.res_id).image_1920 = attachment.datas
+        except Exception as e:
+            _logger.error(
+                "Error while recovering for %s: %s",
+                attachment.res_id,
+                repr(e),
+            )
```
Comment (yostashiro) [inline ir_attachment_conversion_queue_job/models/ir_attachment.py]: Is there actually any point doing `try` and `except` in a job?

```suggestion
        self.env[model].browse(attachment.res_id).image_1920 = attachment.datas
```

File: ir_attachment_conversion_queue_job/models/ir_attachment.py (hunk 3 above)
Comment (AungKoKoLin1997) [reply ir_attachment_conversion_queue_job/models/ir_attachment.py]: I was thinking if we start another migrations (sst15test3, production), there will not be processed ones. That's why I didn't think about excluding.

File: ir_attachment_conversion_queue_job/models/ir_attachment.py [hunk 5]
```diff
@@ -3,14 +3,16 @@
...
-from odoo import models
+from odoo import fields, models
 
 _logger = logging.getLogger(__name__)
 
 
 class Irattachemnt(models.Model):
     _inherit = "ir.attachment"
 
+    is_done = fields.Boolean()
```
Comment (yostashiro) [inline ir_attachment_conversion_queue_job/models/ir_attachment.py]:
```suggestion
    image_conversion_done = fields.Boolean()
```
To be more explicit.

**Review by yostashiro — COMMENTED** [review]
My comments are patchy, but I think the module name should be ir_attachment_image_conversion.

**Pull request closed without merging**